extensions/urban-dictionary/src/index.tsx

```diff
@@ -139,156 +101,69 @@ function ItemDetails({ term }: { term: string }) {
         ))}
       </List.Section>
     </List>
-  )
+  );
 }
 
 function makeLinks(string: string) {
   return string.replace(
     /\[(.*?)\]/gm,
-    (match, term) =>
-      `[${term}](https://www.urbandictionary.com/define.php?term=${term.replace(
-        /\s/g,
-        '+'
-      )})`
-  )
+    (match, term) => `[${term}](https://www.urbandictionary.com/define.php?term=${term.replace(/\s/g, "+")})`
+  );
 }
 
 function getMarkdown(result: UrbanDefineResponseItem) {
   return `
 ${makeLinks(result.definition)}
-> ${makeLinks(result.example.replace(/\r?\n/gm, '\n> '))}
-`
+> ${makeLinks(result.example.replace(/\r?\n/gm, "\n> "))}
+`;
 }
 
-type Endpoint = 'define' | 'autocomplete-extra'
+type Endpoint = "define" | "autocomplete-extra";
 
-function useSearch<T extends Endpoint>(endpoint: T, initial = '') {
-  const [state, setState] = useState<SearchState<T>>({
-    results: [],
-    isLoading: true,
-  })
-  const cancelRef = useRef<AbortController | null>(null)
+function useSearch<T extends Endpoint>(endpoint: T, initial = "") {
+  const params = new URLSearchParams({ term: initial });
+  return useFetch(`https://api.urbandictionary.com/v0/${endpoint}?${params.toString()}`, {
+    method: "GET",
+    parseResponse: async (response) => {
+      const json = (await response.json()) as UrbanResponseList<typeof endpoint>;
```

Comment:
if (!response.ok) {
        throw new Error(response.statusText);
      }

should move before converting to json.

@pernielsentikaer I am not sure how to pull your commit to my local fork, do u mind fixing it ?

Reply:
Should be better now, do you mind checking 🙂

Reply:
lgtm, thanks.